File: php/83-remove-duplicates-from-sorted-list.php
<?php
/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val) { $this->val = $val; }
 * }
 */
class Solution 
{
    /**
     * @param ListNode $head
     * @return ListNode
     */
    public function deleteDuplicates($head) 
    {
        $cur = $head;
        while ($cur != null) {
            $temp = $cur;
            while ($temp->next != null && $temp->next->val == $cur->val) {
                $temp = $temp->next;
            }
            $cur->next = $temp->next;
            $cur = $cur->next;
        }
        return $head;
    }
}

File: go/83-remove-duplicates-from-sorted-list.go
package main

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
	if head == nil {
		return head
	}
	dummy := &ListNode{Next: head}
	node := dummy.Next
	for node.Next != nil {
		val := node.Val
		if val == node.Next.Val {
			temp := node.Next
			for temp != nil && temp.Val == val {
				temp = temp.Next
			}
			node.Next = temp
		} else {
			node = node.Next
		}
	}
	return dummy.Next
}
